compiler/rustc_mir_dataflow/src/impls/liveness.rs

```diff
@@ -129,6 +129,7 @@ where
 #[derive(Eq, PartialEq, Clone)]
 enum DefUse {
     Def,
+    MaybeDef,
```

Comment:
To determine whether an occurrence of a place is a def or a use we need to know the projections, so I would modify `DefUse::for_place` to take both a place and a context. This would render `DefUse::MaybeDef` unnecessary.